modules/dnn/test/test_darknet_importer.cpp

```diff
@@ -125,13 +125,21 @@ class Test_Darknet_nets : public DNNTestLayer
                           const std::vector<int>& refClassIds,
                           const std::vector<float>& refConfidences,
                           const std::vector<Rect2d>& refBoxes,
-                          double scoreDiff, double iouDiff, float confThreshold = 0.24)
+                          double scoreDiff, double iouDiff, float confThreshold = 0.24, int batch_size = 1)
     {
         checkBackend();
 
         Mat sample = imread(_tf("dog416.png"));
-        Mat inp = blobFromImage(sample, 1.0/255, Size(416, 416), Scalar(), true, false);
-
+        Mat inp;
+        if (batch_size > 1){
+            // batch_size matrices with value sample
+            std::vector<Mat> samples(batch_size, sample);
+            InputArrayOfArrays images = InputArrayOfArrays(samples);
+            inp = blobFromImages(images, 1.0/255, Size(416, 416), Scalar(), true, false);
```

Comment:
You can reduce this condition by
```cpp
std::vector<Mat> samples(batch_size, sample);
inp = blobFromImages(samples, 1.0/255, Size(416, 416), Scalar(), true, false);
```
because both `blobFromImage` and `blobFromImages` return 4-dimensional blob.